polyfactory/factories/msgspec_factory.py

```diff
@@ -58,7 +58,7 @@ def get_model_fields(cls) -> list[FieldMeta]:
         fields_meta: list[FieldMeta] = []
 
         for field in type_info.fields:
-            annotation = cls.__model__.__annotations__[field.name]
+            annotation = cls._get_annotation(cls.__model__, field.name)
```

Comment:
use the function - `get_type_hints` from `typing_extensions:

`
```suggestion
            annotation = get_type_hints(cls.__model__, include_extras=True)[field.name]
```